frontend_tests/node_tests/voting_widget.js

```diff
@@ -76,4 +83,202 @@ run_test('poll_data_holder my question', () => {
         question: 'best plan?',
     });
 
+    const invalid_vote_event = {
+        type: 'vote',
+        key: '98,1',
+        vote: 1,
+    };
+
+    blueslip.error = (msg) => {
+        assert.equal(msg, `unknown key for poll: ${invalid_vote_event.key}`);
+    };
+    data_holder.handle_event(sender_id, invalid_vote_event);
+    data = data_holder.get_widget_data();
+
+    const comment_outbound_event = data_holder.handle.new_comment.outbound('new comment');
+    assert.deepEqual(comment_outbound_event, {
+        type: 'new_comment',
+        idx: 2,
+        comment: 'new comment',
+    });
+
+    const new_question = 'Any new plan?';
+    const question_outbound_event = data_holder.handle.question.outbound(new_question);
+    assert.deepEqual(question_outbound_event, {
+        type: 'question',
+        question: new_question,
+    });
+
+    const vote_outbound_event = data_holder.handle.vote.outbound('99,1');
+    assert.deepEqual(vote_outbound_event, { type: 'vote', key: '99,1', vote: -1 });
+
+    vote_event = {
+        type: 'vote',
+        key: '99,1',
+        vote: -1,
+    };
+
+    data_holder.handle_event(sender_id, vote_event);
+    data = data_holder.get_widget_data();
+
+    assert.deepEqual(data, {
+        comments: [
+            {
+                comment: 'release now',
+                names: '',
+                count: 0,
+                key: '99,1',
+            },
+        ],
+        question: 'Any new plan?',
+    });
+});
+
+run_test('activate', () => {
+    /* Testing poll of another person */
+    people.is_my_user_id = return_false;
+    templates.render = (template_name) => {
+        if (template_name === 'poll-widget') {
+            return 'poll-widget';
+        } else if (template_name === 'poll-widget-results') {
+            return 'poll-widget-results';
+        }
+    };
+
+    const widget_elem = $('<div>').addClass('widget-content');
+    let out_data;
+    const callback = (data) => {
+        out_data = data;
+    };
+    const opts = {
+        elem: widget_elem,
+        callback: callback,
+        message: {
+            sender_id: 100,
+        },
+        extra_data: {
+            question: '',
```

Comment:
I think we should also add a test for when the `extra_data.question` is not `''`. We should check if this is set in `poll_question_header`.

Reply:
Sure, I'll add the test for this when user is creating the widget.